pkg/cmd/roachtest/tests/connection_latency.go

```diff
@@ -145,3 +151,392 @@ func registerConnectionLatencyTest(r registry.Registry) {
+) {
+
+	// Create a ldif file to be used to modify the LDAP config
+	// containing all the certificate information for TLS.
+	// each of the certificates have action as `replace`
+	// which would update if already present else add if the
+	// certificates were already present.
+	updateCertCmd := `cat <<EOF | sudo tee /etc/ldap/tls.ldif
+dn: cn=config
+changetype: modify
+replace: olcTLSCACertificateFile
+olcTLSCACertificateFile: /etc/ssl/certs/ca.crt
+-
+replace: olcTLSCertificateKeyFile
+olcTLSCertificateKeyFile: /etc/ssl/certs/node.key
+-
+replace: olcTLSCertificateFile
+olcTLSCertificateFile: /etc/ssl/certs/node.crt
+EOF`
+	err := c.RunE(ctx, option.WithNodes(nodeListOptions), updateCertCmd)
+	require.NoError(t, err)
+
+	// Use the above created file to modify the ldap config.
+	ldapModifyCmd := "sudo ldapmodify -Y EXTERNAL -H ldapi:/// -f /etc/ldap/tls.ldif"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), ldapModifyCmd)
+	require.NoError(t, err)
+
+	// Configure the Uncomplicated Firewall (UFW) to allow incoming
+	// connections on port 636
+	enableFirewallCmd := "sudo ufw allow 636"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), enableFirewallCmd)
+	require.NoError(t, err)
+
+	// Update the file /etc/default/slapd, to contain
+	// SLAPD_SERVICES="ldap:/// ldaps:/// ldapi:///"
+	// this enables connection over LDAP with TLS.
+	updateSlapdConfigCmd := "sudo sed -i 's/^SLAPD_SERVICES=.*/" +
+		"SLAPD_SERVICES=\"ldap:\\/\\/\\/ ldapi:\\/\\/\\/ " +
+		"ldaps:\\/\\/\\/\"/' /etc/default/slapd"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), updateSlapdConfigCmd)
+	require.NoError(t, err)
+
+	// Copy the CA certificate to `/usr/local/share/ca-certificates/`
+	// to add it as a trusted CA for the host node.
+	trustedCACertCmd := "sudo cp /etc/ssl/certs/ca.crt /usr/local/share/ca-certificates/"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), trustedCACertCmd)
+	require.NoError(t, err)
+
+	// Add the custom CA created before as a trusted CA cert for the host node.
+	updateCACertCmd := "sudo update-ca-certificates"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), updateCACertCmd)
+	require.NoError(t, err)
+
+	// Restart the slapd service running in the background
+	// to apply all the configurations changed above.
+	restartSlapdCmd := "sudo systemctl restart slapd"
+	err = c.RunE(ctx, option.WithNodes(nodeListOptions), restartSlapdCmd)
+	require.NoError(t, err)
+
+	// Test the ldap connection over TLS
+	cmd := "sudo ldapsearch -H ldaps://localhost -x -b \"\" -s base -LLL -d 1"
+	_, err = c.RunWithDetailsSingleNode(ctx, t.L(),
+		option.WithNodes(c.Node(1)), cmd)
+	require.NoError(t, err)
+}
+
+// Create the directory structure for openLDAP and test LDAP bind
+// using the `ldapwhoami` command.
+func importUserAndGroupsToLDAPServer(
+	ctx context.Context, t test.Test, c cluster.Cluster, nodeListOptions option.NodeListOption,
+) {
+	// Copy the base structure file onto the node
+	content := getTestDataFileContent(
```

Comment:
Nit: note that `testdata` is for unit (local) tests. This will fail when the `roachtest` binary is executed outside of the project directory. Since this is an implicit dependency (i.e., not enforced by `roachtest`), it's worth mentioning in the description comment for the test.